how-to-release.md

```diff
@@ -0,0 +1,274 @@
+```
+
+## Voting process
+The community including the release manager will be responsible for verifying the release candidate. Anyone can vote +1 or -1 on the release candidates, which helps to gain more confidence on the release. The vote happens on the release Slack channel. The necessary conditions for the release vote to pass are shown [here](https://github.com/pravega/.github/blob/main/governance.md#actions).
+
+There are a few things to check including
+
+- No open release-blocking issues
+- All the unit and integration tests pass
+- Checksums of the artifacts and source code are correct
+- Verify that it can locate, build and run with the staging artifacts. For example, use the [Pravega samples](https://github.com/pravega/pravega-samples) to verify.
+
+It is possible that a release candidate is problematic or denied. In this case, you need to announce the cancellation of this vote, and then fixes issues and repeats the cycle.
+
+Once the community passes the vote, you need to announce the vote passing in the channel.
+
+# Fix any issues
+Any issues identified during the community review and vote should be fixed in this step.
+
+Code changes should be proposed as standard pull requests to the `master` branch and reviewed using the normal contributing process. Then, relevant changes should be cherry-picked into the release branch. The cherry-pick commits should then be proposed as the pull requests against the release branch, again reviewed and merged using the normal contributing process.
+
+Once all issues have been resolved, you should go back and build a new release candidate with these changes.
+
+# Finalize the release
+
+Once the release candidate has been reviewed and approved by the community, the release should be finalized. This involves the github release announcement, deployment of the jar artifacts on maven central, document updates, official docker images publish, etc.
+
+## Step 1: Tagging
+```
+# Checkout to the latest successful-voted tag
+git checkout v0.3.0-rc2
+# Tag this existing tag again as a formal release tag `v0.3.0` and push
+git tag v0.3.0
+git push origin v0.3.0
+```
+
+## Step 2: Create the release page on GitHub
+
+On the GitHub repository page, go to "Releases" and create a new draft. On the draft, fill out the tag field with the correct tag and do NOT check the pre-release box. There are two sections needs to be filled in this github page.
+- Release notes
+- Release assets
+
+### Release notes
+Some introductory text, highlighting the important changes going in the release. This should be reviewed earlier by the community.
+
+### Release assets
+
+Run the following commands to build the distribution artifacts
+```
+./gradlew clean distribution
+```
+
+The files resulting from the build will be under `build/distributions`. For each one of the `.zip`, `.tgz` or `.jar` files in that directory, generate checksums (currently `md5`, `sha1`, and `sha256`). It is easy to do with this bash script.
+```
+#!/bin/bash
```

Comment:
we should commit this script somewhere and do not let people copy paste

Reply:
Is there any reason you want to not let people to copy-paste? My main idea is that most of the scripts I put in this doc can be reused or at least be as a reference, so some more explanation would help me better understand what yor concern is.